pennylane/templates/subroutines/arithmetic/phase_adder.py

```diff
@@ -291,18 +328,24 @@ def _add_k_fourier_loop(i, _k):
 
     aux_k = x_wires[0]
     _add_k_fourier_loop(k)
-    adjoint(_add_k_fourier_loop)(mod)
-    adjoint(QFT)(wires=x_wires)
-    CNOT(wires=[aux_k, work_wire[0]])
-    QFT(wires=x_wires)
-    ctrl(_add_k_fourier_loop, control=work_wire)(mod)
-    adjoint(_add_k_fourier_loop)(k)
-    adjoint(QFT)(wires=x_wires)
-    X(aux_k)
-    CNOT(wires=[aux_k, work_wire[0]])
-    X(aux_k)
-    QFT(wires=x_wires)
-    _add_k_fourier_loop(k)
+    ops.adjoint(_add_k_fourier_loop)(mod)
+    ops.change_op_basis(
+        ops.adjoint(QFT)(wires=x_wires),
+        ops.CNOT(wires=[aux_k, work_wire[0]]),
+        QFT(wires=x_wires),
+    )
+    ops.ctrl(_add_k_fourier_loop, control=work_wire)(mod)
+    ops.change_op_basis(
+        ops.prod(
+            ops.X(aux_k),
+            ops.adjoint(QFT)(wires=x_wires),
+            *reversed(ops.adjoint(_add_k_fourier_loop)(k)),
+        ),
+        ops.CNOT(wires=[aux_k, work_wire[0]]),
+        ops.prod(
+            *make_qscript(_add_k_fourier_loop)(k).operations, QFT(wires=x_wires), ops.X(aux_k)
```

Comment:
Ideally we don't make a quantum script inside a quantum function. Potentially
```suggestion
            *ops.prod(_add_k_fourier_loop)(k), QFT(wires=x_wires), ops.X(aux_k)
```